### PR TITLE
♻️🚮  Upgrade amp-story-bookend to extend from BaseElement 

### PR DESCRIPTION
**Pull request by Enriqe**
(Part of #12167)

* Upgrades amp-story-bookend to extend from BaseElement.
* Deletes code used in v0.1 for rendering articles using `simple-template`. (Part of #14591).
  * We will eventually move everything from `simple-template` to static HTML template helper. #14659.
* In the future we might consider using the store state service so that `amp-story.js` doesn't have to do any calls to `amp-story-bookend.js`.

### CODE REVIEW

File: extensions/amp-story/1.0/bookend/amp-story-bookend.js [hunk 1]
```diff
@@ -13,6 +13,7 @@
  * See the License for the specific language governing permissions and
  * limitations under the License.
  */
+
```
Comment (newmuis) [inline extensions/amp-story/1.0/bookend/amp-story-bookend.js]: nit: is this intended?

Comment (Enriqe) [reply extensions/amp-story/1.0/bookend/amp-story-bookend.js]: The whiteline between the imports and license comment? We have it like that on most files I think

**Comment by Enriqe** [comment]
@gmajoulet @alanorozco please take a look when you have a chance. Thanks!

File: extensions/amp-story/1.0/bookend/amp-story-bookend.js [hunk 2]
```diff
@@ -379,10 +302,10 @@ export class Bookend {
    * will prerender the bookend DOM, but there are cases where we need it before
    * the component is built. Eg: the desktop share button needs the providers.
    * @param {boolean=} applyConfig  Whether the config should be set.
-   * @return {!Promise<(?BookendConfigDef|?./bookend-component.BookendDataDef)>}
+   * @return {!Promise<?./bookend-component.BookendDataDef>}
    */
   loadConfig(applyConfig = true) {
-    if (this.config_ !== undefined) {
+    if (this.config_) {
```
Comment (alanorozco) [inline extensions/amp-story/1.0/bookend/amp-story-bookend.js]: This is getting a little messy. I bet you could simplify a bit (and make Closure complain less) if you split `loadConfig` and `applyConfig` as two separate methods.

`loadConfig` returns a promise (which might already be resolved). `applyConfig` calls `loadConfig`. Simpler and keeps you from checking config state.

File: extensions/amp-story/1.0/bookend/amp-story-bookend.js [hunk 3]
```diff
@@ -269,14 +189,17 @@ export class Bookend {
...
-    this.bookendEl_ = renderAsElement(this.win_.document, ROOT_TEMPLATE);
+    this.root_ = this.win.document.createElement('div');
+    this.bookendEl_ = renderAsElement(this.win.document, ROOT_TEMPLATE);
 
     createShadowRootWithStyle(this.root_, this.bookendEl_, CSS);
 
     this.replayButton_ = this.buildReplayButton_();
 
-    const ampdoc = getAmpdoc(this.parentEl_);
+    let ampdoc;
```
Comment (gmajoulet) [inline extensions/amp-story/1.0/bookend/amp-story-bookend.js]: Now that this inherits from ``AMP.BaseElement``, you can get the ``ampdoc`` using ``this.getAmpDoc()``.
This allows you to delete:
- the ``parentEl_`` property
- the ``buildCallback`` method
- and these four lines : )

File: extensions/amp-story/1.0/bookend/amp-story-bookend.js [hunk 4]
```diff
@@ -593,14 +494,17 @@ export class Bookend {
    * @private
    */
   getStoryMetadata_() {
-    const ampdoc = getAmpdoc(this.parentEl_);
+    let ampdoc;
```
Comment (gmajoulet) [inline extensions/amp-story/1.0/bookend/amp-story-bookend.js]: Same as above: ``this.getAmpDoc()``

File: extensions/amp-story/1.0/bookend/amp-story-bookend.js (hunk 2 above)
Comment (Enriqe) [reply extensions/amp-story/1.0/bookend/amp-story-bookend.js]: sounds good!

File: extensions/amp-story/1.0/bookend/amp-story-bookend.js (hunk 3 above)
Comment (Enriqe) [reply extensions/amp-story/1.0/bookend/amp-story-bookend.js]: nice, thanks!

File: extensions/amp-story/1.0/amp-story.js [hunk 5]
```diff
@@ -1345,10 +1345,12 @@ export class AmpStory extends AMP.BaseElement {
       return Promise.resolve(true);
     }
 
-    return this.bookend_.loadConfig(false /** applyConfig */).then(
-        config => {
-          return !!(config && config.components && config.components.length);
-        });
+    return this.bookend_
+        .loadConfigAndMaybeRenderBookend(false /** renderBookend */).then(
+            config => {
```
Comment (alanorozco) [inline extensions/amp-story/1.0/amp-story.js]: 1. Collapse return into arrow func.
2. Check for actual length 

```js
config => !!(config && config.components && config.components.length > 0)
```

File: extensions/amp-story/1.0/test/test-amp-story-bookend.js [hunk 6]
```diff
@@ -75,8 +76,11 @@ describes.realWin('amp-story-bookend', {
     storyElem = win.document.createElement('amp-story');
     storyElem.appendChild(win.document.createElement('amp-story-page'));
     win.document.body.appendChild(storyElem);
-    story = new AmpStory(storyElem);
-    bookend = new Bookend(win, story.element);
+    new AmpStory(storyElem);
```
Comment (gmajoulet) [inline extensions/amp-story/1.0/test/test-amp-story-bookend.js]: Just wondering: do you know why we need to create a story here?

Comment (Enriqe) [reply extensions/amp-story/1.0/test/test-amp-story-bookend.js]: Woops, didn't need it anymore - it was used for a previous version of the test.

**Pull request merged**